tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+
+### Out of Scope for Scenario 2
+
+- Outer-loop activities such as committing changes, creating PRs, release management, or package publishing
+- GA releases
+- Architectural review preparation
+- Breaking change reviews
+- Updating changelog/`README.md`/metadata content for data-plane libraries
+- Error resolution assistance beyond environment setup remediation
+- Platform support beyond Windows
+- Editors other than VS Code
+
+---
+
+## Workflow
+
+**Legend**: 🆕 = New tool for Scenario 2 | Existing tool from Scenario 1
+
+1. **Environment Setup** → `azsdk_verify_setup`, 🆕 `azsdk_package_setup_env`
+   - Verify tools and versions (see [Scenario 1 – Environment Setup](./0-scenario-1.spec.md#1-environment-setup))
+   - Remediate missing or out-of-date tools with azsdk_package_setup_env
+
+2. **Generating** → `azsdk_package_generate_code`
+   - Generate SDK code, tests, and samples (see [Scenario 1 – Generating](./0-scenario-1.spec.md#2-generating))
+   - **Note**: Generation tooling now handles library project bootstrapping for [Net-New SDKs](#net-new-sdk)
+
+3. **Determine Customization Approach** → 🆕 `azsdk_package_customization_playbook`
+   - Consult playbook to determine whether customizations should be done via TypeSpec or code
+   - Get decision criteria and rationale based on Azure SDK guidelines
+
+4. **TypeSpec Customizations** → 🆕 `azsdk_package_customize_typespec`
+   - Modify client.tsp
+   - Validate TypeSpec correctness
+
+5. **Code Customizations** → 🆕 `azsdk_package_customized_code_update`
+   - Add handwritten/custom code on top of generated output
+
+6. **Testing** → `azsdk_package_run_tests`, 🆕 `azsdk_package_test_resources_manage`, 🆕 `azsdk_package_test_assets_create`, 🆕 `azsdk_package_test_mode_set`
+
+   If net-new SDK:
+   - Create test asset files with `azsdk_package_test_assets_create`
+   - Set up test infrastructure and Azure resources with `azsdk_package_test_resources_manage --action create`
+   - Set testing mode to live with recording: `azsdk_package_test_mode_set --mode live-record`
+   - Run tests with `azsdk_package_run_tests`
+   - Tear down Azure resources with `azsdk_package_test_resources_manage --action delete`
+   - Set testing mode to playback: `azsdk_package_test_mode_set --mode playback`
+   - Run playback tests with `azsdk_package_run_tests`
+
+   If existing SDK:
+   - Set testing mode to live: `azsdk_package_test_mode_set --mode live`
+   - Run live tests with `azsdk_package_run_tests`
+   - Re-record failing or outdated tests
+   - Tear down temporary resources with `azsdk_package_test_resources_manage --action delete`
+   - Run playback tests with `azsdk_package_run_tests`
+
+7. **Update Package/Docs/Metadata** → `azsdk_package_update_metadata`, `azsdk_package_update_version`, `azsdk_package_update_changelog_content`
+   - Update package metadata, docs, and changelogs (see [Scenario 1 – Update Package/Docs/Metadata](./0-scenario-1.spec.md#3-update-packagedocsmetadata))
```

Comment:
For net SDK scenario:
- we will add a new tool `azsdk_package_update_ci` to provision & update `ci.yml` file.
- the `azsdk_package_update_metadata` tool needs to support parent level and root level `pom.xml` update for Java libraries.

We will extend `azsdk_package_update_version` tool to update version for data plane libraries. In scenario 1, the tool returns `no-op` for data plane libraries.